router/core/src/main/scala/io/buoyant/router/ctx.scala

```diff
@@ -0,0 +1,55 @@
+package io.buoyant.router
+
+import com.twitter.finagle._
+import com.twitter.finagle.buoyant.Dst
+import com.twitter.finagle.context.Contexts
+import com.twitter.util.Local
+
+package object ctx {
+
+  private class SetFilter[T, Req, Rsp](key: Contexts.local.Key[T], value: T)
+    extends SimpleFilter[Req, Rsp] {
+
+    def apply(req: Req, service: Service[Req, Rsp]) = {
+      val save = Local.save()
```

Comment:
Is this right?  Doesn't `let` already take care of saving and restoring?  Isn't the thread local referred to by `Local.save` actually different from `Contexts.local`?